### ISSUE TITLE
Assertion `Mask.size() == TpNumElts && "Expected Mask and Tp size to match!"' failed, since "[SLP]Improve costs in computeExtractCost()"

### ISSUE THREAD

**Issue by mstorsjo** [llvm/llvm-project#67933]
Since 019aee832768b0704048905cd8ce20299e7e0b25 (where the change was relanded) or the original 59a67ea35d608480257fc64ec3e5106ef50de740, the following snippet triggers failed asserts for the aarch64 targets:
```c
int b;                                  
int *c; 
void d(double *e, double *i, double *l) { 
  for (int f = 0; f < 3; f++)
    for (int g = 0; g < 3; g++) {
      e[f * 3 + g] = 0.;
      for (int h = 0; h < 3; h++)
        e[f * 3 + g] += i[f + h] * l[h * 3 + g];
    }   
}       
void p() {
  double j[] = {149.68500, 29.07000, 43.02765, 84.47235, 20.73466};
  double k[] = {257.00392, 0.00000,   360.31950, 257.00392, 88.44438,
                183.53583, 257.00392, 455.41095, 0.00000},
         n[9], o[9];
  double a = b / 255.;
  double m[] = {a, a * b, a, a * b, b};         
  d(n, m, k);
  d(o, j, n);                           
  for (int f = 0; f < 9; f++)
    c[f] = o[f];                        
}       
```
Compiled like this:
```
$ clang -target aarch64-linux-gnu -c extract-reduced.c -O2 -funsafe-math-optimizations
clang: ../lib/Target/AArch64/AArch64TargetTransformInfo.cpp:3565: llvm::InstructionCost llvm::AArch64TTIImpl::getShuffleCost(llvm::TargetTransformInfo::ShuffleKind, llvm::VectorType*, llvm::ArrayRef<int>, llvm::TargetTransformInfo::TargetCostKind, int, llvm::VectorType*, llvm::ArrayRef<const llvm::Value*>): Assertion `Mask.size() == TpNumElts && "Expected Mask and Tp size to match!"' failed.
```

CC @alexey-bataev 

I'd like to revert this change if it takes long to fix.

**Comment by alexey-bataev** [comment]
Thanks, will fix it tomorrow

**Comment by alexey-bataev** [comment]
Fixed in 263a00fa910bd9d19c8590ad9f7ae1bdd9d76c02

**Comment by mstorsjo** [comment]
> Fixed in [263a00f](https://github.com/llvm/llvm-project/commit/263a00fa910bd9d19c8590ad9f7ae1bdd9d76c02)

Thanks! I can confirm that the full original case seems to build fine now as well.
